framework/PostProcessors/DataMining.py

```diff
@@ -213,159 +214,175 @@ def _localGenerateAssembler(self,initDict):
     """
     self.jobHandler = initDict['internal']['jobHandler']
 
-  def inputToInternal(self, currentInp):
+  def inputToInternalForHistorySet(self,currentInput):
     """
-      Function to convert the received input into a format this object can
-      understand
-      @ In, currentInp, list or DataObjects, Some form of data object or list of
-        data objects handed to the post-processor
-      @ Out, inputDict, dict, An input dictionary this object can process
+      Function to convert the input history set into a format that this
+      post-processor can understand
+      @ In, currentInput, object, DataObject of currentInput
+      @ Out, inputDict, dict, an input dictionary that this post-processor can process
     """
-
-    if type(currentInp) == list:
-      currentInput = currentInp[-1]
-    else:
-      currentInput = currentInp
-
-    if currentInput.type == 'HistorySet' and self.PreProcessor is None and self.metric is None:
-      # for testing time dependent dm - time dependent clustering
-      inputDict = {'Features':{}, 'parameters':{}, 'Labels':{}, 'metadata':{}}
-
-      # FIXME, this needs to be changed for asynchronous HistorySet
-      if self.pivotParameter in currentInput.getParam('output',1).keys():
-        self.pivotVariable = currentInput.getParam('output',1)[self.pivotParameter]
-      else:
-        self.raiseAnError(ValueError, 'Pivot variable not found in input historyset')
-      # end of FIXME
-
-      historyKey = currentInput.getOutParametersValues().keys()
-      numberOfSample = len(historyKey)
-      numberOfHistoryStep = len(self.pivotVariable)
+    dataSet = currentInput.asDataset()
+    inputDict = {'Features': {}, 'parameters': {}, 'Labels': {}, 'metadata': {}}
+    if self.pivotParameter is None and self.metric is not None:
+      if not hasattr(self.metric, 'pivotParameter'):
+        self.raiseAnError(IOError, 'HistorySet is provided as input, but this post-processor ', self.name, ' can not handle it!')
+      self.pivotParameter = self.metric.pivotParameter
+    if self.PreProcessor is None and self.metric is None:
+      if not currentInput.checkIndexAlignment(indexesToCheck=self.pivotParameter):
+        self.raiseAnError(IOError, "The data provided by the DataObject ", currentInput.name, " is not synchronized!")
+      # for testing time dependent data mining - time dependent clustering
+      self.pivotVariable = np.asarray([dataSet.isel(RAVEN_sample_ID=i)[self.pivotParameter].dropna(self.pivotParameter).values for i in range(len(currentInput))])
```

Comment:
Instead to use the ```RAVEN_sample_ID``` directly, you should inquire the dataobject ```currentInput.sampleTag```

Reply:
fixed